app/services/provider_interface/reference_status_report_service.rb

```diff
@@ -0,0 +1,57 @@
+module ProviderInterface
+  class ReferenceStatusReportService
+    attr_reader :provider_application_references, :context
+
+    HEADERS = [
+      'Name',
+      'Status',
+      'Referee Name',
+      'Application Choice ID',
+      'Link',
+    ].freeze
+
+    def initialize(provider_application_references, context:)
+      @provider_application_references = provider_application_references
+      @context = context
+    end
+
+    def headers
+      HEADERS
+    end
+
+    def rows
+      provider_application_references.map do |provider_application_reference|
+        {
+          header: full_name(provider_application_reference),
+          values: [
+            feedback_status_label(provider_application_reference.feedback_status),
+            provider_application_reference.name,
+            provider_application_reference.application_choice_id,
+            reference_link(provider_application_reference),
+          ],
+        }
+      end
+    end
+
+    def full_name(reference)
+      "#{reference.first_name} #{reference.last_name}"
+    end
+
+    def reference_link(reference)
+      {
+        link: context.provider_interface_application_choice_references_path(reference.application_choice_id),
+        text: 'Application',
+      }
+    end
+
+    def feedback_status_label(status)
+      if status == 'feedback_requested'
+        'Sent'
+      elsif status == 'feedback_provided'
+        'Received'
+      else
+        'Other'
+      end
+    end
+  end
+end
```

Comment:
Can we move the strings to a locale?
e.g config/locales/reference_status_report_service.yml where you add the headers and the statuses